packages/wdio-reporter/tests/__mocks__/@wdio/reporter.js

```diff
@@ -24,6 +25,47 @@ export default class WDIOReporter extends EventEmitter {
         return true
     }
 
+    getEventsToReport (suite) {
+        return [
+            /**
+             * report all tests
+             */
+            ...suite.tests,
+            /**
+             * and only hooks that failed
+             */
+            ...suite.hooks
+                .filter((hook) => Boolean(hook.error))
+        ]
+    }
+
+    getFailureDisplay () {
+        let failureLength = 0
+        const output = []
+
+        for (const [ , suite ] of Object.entries(this.all_suites)) {
+            const suiteTitle = suite.title
+            const eventsToReport = this.getEventsToReport(suite)
+            for (const test of eventsToReport) {
+                if(test.state !== 'failed') {
+                    continue
+                }
+
+                const testTitle = test.title
+
+                // If we get here then there is a failed test
+                output.push(
+                    '',
+                    `${++failureLength}) ${suiteTitle} ${testTitle}`,
+                    chalk.red(test.error.message),
+                    ...test.error.stack.split(/\n/g).map(value => chalk.gray(value))
+                )
+            }
+        }
+
+        return output
+    }
```

Comment:
We should not reimplement functionality in a mock. We are just interested that it is executed with the right params and that it returns something

Reply:
I can maybe unmock the reporter for those tests.

Reply:
Why, it should stay mocked. We want to isolate our unit tests.

Reply:
I need to be able to verify the display and output of that method for the report so the mock won't work that way. I think I can maybe just create a util function with the output, call that in getFailureDisplay and unit test the util function for the report display. Then I can mock the call to getFailureDisplay.

Reply:
You should test `getFailureDisplay` as part of https://github.com/webdriverio/webdriverio/blob/master/packages/wdio-reporter/tests/reporter.test.js and then stub the methods in the mock and check that these are called.

Reply:
oh duh, I'm an idiot :)